File: examples/external-clients/nodejs/producer.js
// Produce messages to Streamline using KafkaJS (standard Kafka client).

const { Kafka } = require("kafkajs");

const brokers = (process.env.KAFKA_BOOTSTRAP_SERVERS || "localhost:9092").split(",");
const kafka = new Kafka({ brokers });
const producer = kafka.producer();

async function main() {
  await producer.connect();

  for (let i = 0; i < 5; i++) {
    const msg = JSON.stringify({ event: "test", index: i });
    const result = await producer.send({
      topic: "demo",
      messages: [{ value: msg }],
    });
    const { partition, baseOffset } = result[0];
    console.log(`Sent to partition=${partition} offset=${baseOffset}: ${msg}`);
  }

  await producer.disconnect();
  console.log("Done! 5 messages produced to 'demo' topic.");
}

main().catch(console.error);

File: examples/external-clients/python/producer.py
"""Produce messages to Streamline using kafka-python (standard Kafka client)."""

import json
import os

from kafka import KafkaProducer

bootstrap = os.environ.get("KAFKA_BOOTSTRAP_SERVERS", "localhost:9092")

producer = KafkaProducer(
    bootstrap_servers=bootstrap,
    value_serializer=lambda v: json.dumps(v).encode("utf-8"),
)

for i in range(5):
    msg = {"event": "test", "index": i}
    future = producer.send("demo", value=msg)
    metadata = future.get(timeout=10)
    print(f"Sent to partition={metadata.partition} offset={metadata.offset}: {msg}")

producer.flush()
producer.close()
print("Done! 5 messages produced to 'demo' topic.")
